From 0603407e9467411a65f07ba12b81b8ecf370dca4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E7=8E=8B=E6=8C=AF=E9=82=A6?= <wangzhenbang5@huawei.com>
Date: Mon, 7 Apr 2025 10:18:20 +0800
Subject: [PATCH] change MoveTo interface

---
 mindformers/modules/transformer/moev3.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mindformers/modules/transformer/moev3.py b/mindformers/modules/transformer/moev3.py
index 210db93d..a3b47455 100644
--- a/mindformers/modules/transformer/moev3.py
+++ b/mindformers/modules/transformer/moev3.py
@@ -44,7 +44,7 @@ from mindformers.modules.transformer.moe import default_moe_config
 from mindformers.version_control import check_moveto_op_support
 
 if check_moveto_op_support():
-    D2H = ops.gen_ops_prim.MoveTo().add_prim_attr("recompute", False)
+    D2H = ops.MoveTo().add_prim_attr("recompute", False)
 
 
 class MoEV3(Cell):
-- 
2.33.1.windows.1

