diff -Npur cppjieba-v5.1.1/CMakeLists.txt cppjieba-v5.1.1-change/CMakeLists.txt
--- cppjieba-v5.1.1/CMakeLists.txt	2024-06-07 17:19:23.000000000 +0800
+++ cppjieba-v5.1.1-change/CMakeLists.txt	2025-03-06 10:03:47.159513677 +0800
@@ -2,6 +2,10 @@ PROJECT(CPPJIEBA)
 
 CMAKE_MINIMUM_REQUIRED (VERSION 2.6)
 
+if (MSVC)
+    add_compile_options(/utf-8)
+endif()
+
 INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/deps/limonp/include
   ${PROJECT_SOURCE_DIR}/include)
 

